lib/Corres_Cases.thy

```diff
@@ -0,0 +1,319 @@
+(*
+ * Copyright 2023, Proofcraft Pty Ltd
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ *)
+
+theory Corres_Cases
+imports Corres_UL
+begin
+
+text \<open>
+  This file defines the following main methods for safe data type case distinctions on
+  corres/corres_underlying predicates.
+
+  \<^item> corres_cases_left: case distinction on abstract monad
+  \<^item> corres_cases_right: case distinction on concrete monad
+  \<^item> corres_cases: case distinction first of abstract/concrete monad
```

Comment:
I am not clear on what this means here or in the commit message: "case distinction first of abstract/concrete monad"
Does this mean it will apply it to the first side it is able to out of those two, starting with abstract side and if that fails moving on to concrete side? If so, this could be phrased a bit more clearly.

Reply:
That is indeed what it is supposed to mean. Will rephrase.